Lib/fontParts/base/segment.py

```diff
@@ -294,6 +308,8 @@ def _get_offCurve(self):
         """
         Subclasses may override this method.
         """
+        if self.points and self.points[-1].type == "offcurve":
```

Comment:
Given my comment above, I think this addition should not be necessary. Validation may need to be adapted, that if `segment.type == "qcurve"` the `segment.points[-1]` should be allowed to be None.